### PR TITLE
Speed up columns slices

### LINKED ISSUE #775: Speed up columns slices: `etna.datasets.utils.select_columns`
### 🚀 Feature Request

In a lot of places we use `df.loc[:, pd.IndexSlice[segments, column]]` to select `column` from all the segments. It appears to be very slow on a lot of segments.

We should find places where we use it and make sure that it can be replaced with `df.loc[:, pd.IndexSlice[:, column]]` without problems. 

Where was some problem with the second choice: #188. We should investigate is it still existing and in which conditions:
1. Is it applicable for selection only one column? (`SklearnTransform` selects many)
2. Can it be avoided by some trick in taking slices (sorting columns for example).

### Proposal

1. Find all places with slow slice `df.loc[:, pd.IndexSlice[segments, column]]` where column is scalar. Replace them with function (you can add it `etna.datasets.utils`). Try to replace slow slice in function with fast slice: `df.loc[:, pd.IndexSlice[:, column]`. Make sure that in that case we don't have reordering of columns in different pandas versions.
2. Do the same but with list of values in `column` (e.g. `SklearnTransform`) and investigate reordering issue during testing. We want to avoid it without putting all the segments into the slice.
3. Make some benchmarking that changed transforms (or other calls) become faster. Add code for benchmarking and its results in the comments of PR. E.g. you can take dataframe with 50000 segments, 100 timestamps, 5 additional int columns, 5 additional float columns, 5 additional category columns.

### Test cases

1. Make sure that current tests pass for scalar case.
2. Make sure that current tests pass for list case.
4. Add tests on function for selection of one column.
5. Add tests on function for selection of multiple columns (in `SklearnTransform` we had some tests on reordering, it can be useful).

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Mr-Geekman**
## Before submitting (must do checklist)

- [x] Did you read the [contribution guide](https://github.com/tinkoff-ai/etna/blob/master/CONTRIBUTING.md)?
- [x] Did you update the docs? We use Numpy format for all the methods and classes.
- [x] Did you write any new necessary tests?
- [x] Did you update the [CHANGELOG](https://github.com/tinkoff-ai/etna/blob/master/CHANGELOG.md)?

## Proposed Changes
Look #775.

## Closing issues
Closes #775.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -107,3 +107,50 @@ jobs:
...
+        with:
+          python-version: 3.8
+
+      - name: Install Poetry
+        uses: snok/install-poetry@v1
+        with:
+          virtualenvs-create: true
+          virtualenvs-in-project: true
+
+      - name: Load cached venv
```
Comment (martins0n) [inline .github/workflows/test.yml:137]: We shouldn't use cache there -- it could be corrupted.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -107,3 +107,50 @@ jobs:
...
+        id: cached-poetry-dependencies
+        uses: actions/cache@v2
+        with:
+          path: .venv
+          key: venv-${{ runner.os }}-3.8-${{ hashFiles('**/poetry.lock') }}
+
+      - name: Install dependencies
```
Comment (martins0n) [inline .github/workflows/test.yml:144]: image

As a result of caching we will run the same test case without pandas installing.

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -107,3 +107,50 @@ jobs:
...
+        if: steps.cached-poetry-dependencies.outputs.cache-hit != 'true'
+        run: |
+          poetry add "pandas${{ matrix.pandas-version }}"
```
Comment (martins0n) [inline .github/workflows/test.yml:147]: I guess installing via `poetry add` will lead to long dependency resolving.

We should install it via pip after poetry install.

File: make_bug.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
+import numpy as np
```
Comment (martins0n) [inline make_bug.py:1]: Hm, what is the purpose of this script?

File: etna/transforms/math/statistics.py [hunk 5]
```diff
@@ -73,7 +73,8 @@ def transform(self, df: pd.DataFrame) -> pd.DataFrame:
         history = self.seasonality * self.window if self.window != -1 else len(df)
         segments = sorted(df.columns.get_level_values("segment").unique())
 
-        x = df.loc[pd.IndexSlice[:], pd.IndexSlice[segments, self.in_column]].values[::-1]
+        df_slice = df.loc[pd.IndexSlice[:], pd.IndexSlice[:, self.in_column]].sort_index(axis=1)
```
Comment (martins0n) [inline etna/transforms/math/statistics.py:76]: I guess, we don't need `pd.IndexSlice[:]`

**Review by martins0n — CHANGES_REQUESTED** [review]
Maybe we should benchmarks? 
Sorts everywhere could lead to bottlenecks.

**Comment by Mr-Geekman** [comment]
I added report with benchmark into `etna-experiments`, PR: `ISSUE - 775`.

**Review by martins0n — APPROVED** [review]
👍

**Review by martins0n — APPROVED** [review]

**Pull request merged**